client/lib/graph/GraphProvider.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes, Children } from 'react';
+
+export default class GraphProvider extends Component {
+	static propTypes = {
+		graph: PropTypes.object.isRequired,
+		children: PropTypes.element.isRequired
+	};
+
+	static childContextTypes = {
+		graph: PropTypes.object.isRequired
+	};
+
+	getChildContext() {
+		return { graph: this.graph };
+	}
+
+	constructor( props, context ) {
+		super( props, context );
+		this.graph = props.graph;
```

Comment:
Actually, this is a good practice to avoid updating the context (if someone tries to update the graph prop). 

https://facebook.github.io/react/docs/context.html#updating-context